lib/smack/SmackRep.cpp

```diff
@@ -1012,13 +1017,48 @@ std::string SmackRep::code(llvm::CallInst& ci) {
 
   std::string s = fmt;
   for (unsigned i=1; i<ci.getNumOperands()-1; i++) {
-    const Expr* a = arg(f, i, ci.getOperand(i));
+    Value* argV = ci.getOperand(i);
     std::string::size_type idx = s.find('@');
     assert(idx != std::string::npos && "inline code: too many arguments.");
 
+    llvm::Type* targetType = argV->getType();
+    bool isCast = false;
+    if (idx + 1 < s.length()) {
+      switch(s[idx+1]) {
+      case 'c':
+      case 'b':
+      case 'B':
+        targetType = Type::getInt8Ty(argV->getContext());
+        isCast = true;
+        break;
+      case 'f':
```

Comment:
It seems the rational comment is gone, but this is the gist of it.
The cases are meant to match up with the format characters in https://docs.python.org/2.7/library/struct.html#format-characters. Lowercase and capitals are unsigned and signed respectively. Even though LLVM doesn't differentiate between signed-ness, it might be nice to have the type in the format string match the type in the code.
I'll update the doc to reflect this.